### PR TITLE
PLT-4767 Implement MFA Enforcement

### PR DESCRIPTION
**Pull request by jwilander**
#### Summary
Adds MFA enforcement such that, when it is enabled:
* Any logged in user without MFA will be redirected to a page to setup MFA enforcement
* Any logging in user without MFA will be redirected to a page to setup MFA enforcement
* Any newly signed up user will be redirected to a page to setup MFA enforcement
* Navigation to pages outside of MFA setup will be blocked until MFA is added to the account
* Blocks access to user required, non-mfa related APIs until MFA is added to the account
* Also moved MFA to it's own tab in the System Console and added the enforcement option

#### Ticket Link
https://mattermost.atlassian.net/browse/PLT-4767

#### Checklist
- [x] Has UI changes
- [x] Includes text changes and localization file ([.../i18n/en.json](https://github.com/mattermost/platform/blob/master/i18n/en.json) and [.../webapp/i18n/en.json](https://github.com/mattermost/platform/tree/master/webapp/i18n/en.json)) updates
- [x] Touches critical sections of the codebase (auth, upgrade, etc.)
- [x] Has enterprise changes (mattermost/enterprise#74)

### CODE REVIEW

**Comment by jwilander** [comment]
Still needs some UI work from @asaadmahmood  as well

**Comment by esethna** [comment]
+1 on Zapier integration. I think LB is taking the rest of the PM review. I tested:

1. Connecting a zapier account then turning on MFA enforcement.
2. Turning on MFA enforcement then connecting a zapier account.
3. Connecting a zapier account then reseting MFA to a different device.

**Comment by lfbrock** [comment]
+1, confirmed all the changes were fixed. (Overall, have tested on mobile, different browsers, and signing in/switching between email, LDAP, GitLab SSO).

File: config/config.json [hunk 1]
```diff
@@ -21,10 +21,11 @@
         "EnablePostUsernameOverride": false,
         "EnablePostIconOverride": false,
         "EnableTesting": false,
-        "EnableDeveloper": false,
+        "EnableDeveloper": true,
```
Comment (enahum) [inline config/config.json]: Can you revert this file and only leave the new configuration settings?

Comment (jwilander) [inline config/config.json]: Whoops yep, will do

**Comment by jwilander** [comment]
@enahum fixed

File: api/context.go [hunk 2]
```diff
@@ -46,51 +46,55 @@ type Context struct {
 }
 
 func ApiAppHandler(h func(*Context, http.ResponseWriter, *http.Request)) http.Handler {
-	return &handler{h, false, false, true, false, false, false}
+	return &handler{h, false, false, true, false, false, false, false}
```
Comment (coreyhulen) [inline api/context.go]: Next time, lets break this out as named params.  It's getting kind of hard to follow.

**Pull request merged**